tests/chainer_tests/functions_tests/rnn_tests/test_function_n_step_rnn.py

```diff
@@ -23,77 +14,127 @@ def _relu(x):
+            'use_cudnn': ['never'],
+        })
+
+        + testing.product({
+            'use_cudnn': ['always'],
+            'cudnn_deterministic': [True, False],
+            'autotune': [True, False],
+        })]))
+class TestNStepRNN(testing.FunctionTestCase):
+
+    dodge_nondifferentiable = True
 
     def setUp(self):
-        self.xs = _shaped_random([(b, self.in_size) for b in self.batches])
-        h_shape = (self.n_layers, self.batches[0], self.out_size)
-        self.hx = _shaped_random(h_shape)
-
-        o = self.out_size
-        i = self.in_size
-        self.ws = []
-        self.bs = []
-        # The first layer has the different shape
-        self.ws.append(_shaped_random([(o, i), (o, o)]))
-        self.bs.append(_shaped_random([o, o]))
-        for _ in range(self.n_layers - 1):
-            self.ws.append(_shaped_random([(o, o), (o, o)]))
-            self.bs.append(_shaped_random([o, o]))
-
-        self.dys = _shaped_random([(b, self.out_size) for b in self.batches])
-        self.dhy = _shaped_random(h_shape)
-
-    def check_forward(
-            self, h_data, xs_data, ws_data, bs_data):
-        h = _wrap_variable(h_data)
-        xs = _wrap_variable(xs_data)
-        ws = _wrap_variable(ws_data)
-        bs = _wrap_variable(bs_data)
-        hy, ys = functions.n_step_rnn(
-            self.n_layers, self.dropout, h, ws, bs, xs,
-            activation=self.activation)
-
-        e_hy = self.hx.copy()
-        for ind in range(self.length):
-            x = self.xs[ind]
+        self.skip_double_backward_test = True
+
+    def generate_inputs(self):
+        h_shape = (self.n_layers, self.batches[0], self.hidden_size)
+        dtype = numpy.float32
+
+        h = array(h_shape, dtype)
+        in_size = self.input_size
+        out_size = self.hidden_size
+        xs = [array((self.batches[b], in_size), dtype)
+              for b in range(len(self.batches))]
+
+        def w_in(i, j):
+            return in_size if i == 0 and j < 1 else out_size
+
+        inputs = []
+        inputs.append(h)
+        for i in range(len(self.batches)):
+            inputs.append(xs[i])
+        for n in range(self.n_layers):
+            for i in range(2):
+                inputs.append(array((out_size, w_in(n, i)), dtype))
+            for i in range(2):
+                inputs.append(array((out_size,), dtype))
+        return tuple(inputs)
+
+    def process_inputs(self, inputs):
+        h = inputs[0]
+
+        xs = inputs[1: 1 + len(self.batches)]
+        ws = []
+        bs = []
+        index = 1 + len(self.batches)
+        for n in range(self.n_layers):
+            ws.append(inputs[index: index + 2])
+            bs.append(inputs[index + 2: index + 4])
+            index += 4
+
+        return h, ws, bs, xs
+
+    def forward(self, inputs, device):
+        h, ws, bs, xs = self.process_inputs(inputs)
+        if h.array.dtype == numpy.float64:
+            with chainer.using_config('use_cudnn', 'never'):
```

Comment:
The previous test code did not have this workaround.
The problem is that `FunctionTestCase` passes `dtype=float64` to `gradient_check.check_backward()` and there's no way to change this behavior.
I sent a PR to make it configurable: #7817
After #7817 merged, lets avoid this workaround and set `numerical_grad_dtype = None`.

Reply:
Yes, sure!

Reply:
#7817 has been merged!